### PR TITLE
feat: add create payment type component

### PR DESCRIPTION
**Pull request by Rishabh106**
## Description
Added Routing to the Create Payment Type Button, implemented the API and designed its UI

## Related issues and discussion
#164

## Screenshots, if any
create_payment_type

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by abhaychawla** [comment]
@Rishabh106 Please don't work on an issue without commenting on it. This was added as a GCI task.

**Comment by Rishabh106** [comment]
> @Rishabh106 Please don't work on an issue without commenting on it. This was added as a GCI task.

Sorry, I will claim the task once my current task is reviewed.

File: src/app/organization/payment-types/create-payment-type/create-payment-type.component.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            </mat-error>
+          </mat-form-field>
+
+          <mat-form-field>
+            <mat-label>Description</mat-label>
+            <input matInput formControlName="description">
+          </mat-form-field>
+          <span class="span1"></span>
+  
+          <mat-checkbox labelPosition="before" formControlName="isCashPayment">
```
Comment (abhaychawla) [inline src/app/organization/payment-types/create-payment-type/create-payment-type.component.html]: Apply a class to checkbox (only if necessary) with an appropriate name.

File: src/app/organization/payment-types/create-payment-type/create-payment-type.component.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            <input required matInput formControlName="name">
+            <mat-error *ngIf="paymentTypeForm.controls.name.hasError('required')">
+              Payment Type is <strong>required</strong>
+            </mat-error>
+          </mat-form-field>
+
+          <mat-form-field>
+            <mat-label>Description</mat-label>
+            <input matInput formControlName="description">
+          </mat-form-field>
```
Comment (abhaychawla) [inline src/app/organization/payment-types/create-payment-type/create-payment-type.component.html]: Should be a text area, not an input.

File: src/app/organization/payment-types/create-payment-type/create-payment-type.component.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+
+/** Custom Services */
+import { OrganizationService } from '../../organization.service';
+
```
Comment (abhaychawla) [inline src/app/organization/payment-types/create-payment-type/create-payment-type.component.ts]: Missing comment.

File: src/app/organization/payment-types/create-payment-type/create-payment-type.component.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {OrganizationService} organizationService Organization Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   */
+  constructor(private formBuilder: FormBuilder,
+    private organizationService: OrganizationService,
+    private router: Router,
+    private route: ActivatedRoute) {}
+
```
Comment (abhaychawla) [inline src/app/organization/payment-types/create-payment-type/create-payment-type.component.ts]: Missing comment.

File: src/app/organization/payment-types/create-payment-type/create-payment-type.component.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  /**
+   * Creates the payment type form.
+   */
+  createpaymentTypeForm() {
+    this.paymentTypeForm = this.formBuilder.group({
+      'name': ['', Validators.required],
+      'description': [''],
+      'isCashPayment' : [false],
```
Comment (abhaychawla) [inline src/app/organization/payment-types/create-payment-type/create-payment-type.component.ts]:
```suggestion
      'isCashPayment': [false],
```

**Pull request merged**